### DISCUSSION TITLE
New to ojs and want to make histogram of simulated data

### DISCUSSION THREAD

**Discussion by DamonCharlesRoberts** [quarto-dev/quarto-web#611] [Q&A]
Hello!

I am trying to make some simple interactives with simulated data to demonstrate some concepts to students of mine. I thought OJS could be a really useful tool to do this.

However, I am running into troubles.

I first simulate the data with numpy and put it into a pandas.DataFrame
```{python}
from random import seed
from numpy.random import normal, negative_binomial
import pandas as pd

data = pd.DataFrame({
    "Normal": normal(0,1,100000),
    "Negative_binomial": negative_binomial(1, 0, 100000)
})
```

And then I use `ojs_define()`:
```{python}
ojs_define(simulated = data)
```

Then I wanted to make a histogram, but wasn't sure if I got my data in correctly, so then I thought I'd try to produce a table, but still have a blank page:
```{ojs}
data = transpose(simulated)
```

```{ojs}
Inputs.table(dataset)
```

What am I missing?

**Comment by DamonCharlesRoberts** [comment]
Ahhh. Okay, I figured it out. I am dumb.

For anyone who might run into this problem:

I was trying to view the output through the html file it produces only. 

Instead you can run `quarto preview file_with_ojs.qmd`. This will show you everything. It is because it relies on the JS libs that are put into a folder. Opening the html only will not pull those in to help you view the outputs.

**Comment by DamonCharlesRoberts** [comment]
Ahhh. Okay, I figured it out. I am dumb.

For anyone who might run into this problem:

I was trying to view the output through the html file it produces only. 

Instead you can run `quarto preview file_with_ojs.qmd`. This will show you everything. It is because it relies on the JS libs that are put into a folder. Opening the html only will not pull those in to help you view the outputs.

**Reply by cscheid** [reply]
You should check that you're running quarto 1.3, which will warn you about running OJS in `file:///` instead of a local server.

**Reply by DamonCharlesRoberts** [reply]
Ahh. Helpful! Thanks!
